File: backend/app/models/analysis.py
from pydantic import BaseModel
from typing import List, Literal


class TipsCategoryScore(BaseModel):
    category: str
    score: int  # 0-100


class Evaluations(BaseModel):
    technology: int  # 0-100
    business: int    # 0-100
    team: int        # 0-100
    tipsFit: int     # 0-100


class AnalysisResult(BaseModel):
    companySummary: str
    tipsCategories: List[TipsCategoryScore]
    evaluations: Evaluations
    overallScore: int  # 0-100
    recommendation: Literal["추천", "보류", "비추천"]
    strengths: List[str]  # TOP 3
    risks: List[str]      # TOP 3
    comments: str


class AnalysisRequest(BaseModel):
    file_id: str


class UploadResponse(BaseModel):
    file_id: str
    filename: str
    file_size: int

File: frontend/src/types/analysis.ts
export interface TipsCategoryScore {
  category: string;
  score: number; // 0-100
}

export interface Evaluations {
  technology: number; // 0-100
  business: number;   // 0-100
  team: number;       // 0-100
  tipsFit: number;    // 0-100
}

export interface AnalysisResult {
  companySummary: string;
  tipsCategories: TipsCategoryScore[];
  evaluations: Evaluations;
  overallScore: number; // 0-100
  recommendation: "추천" | "보류" | "비추천";
  strengths: string[];
  risks: string[];
  comments: string;
}

export interface UploadResponse {
  file_id: string;
  filename: string;
  file_size: number;
}
